### ISSUE TITLE
[rllib] no improvement APEX-DQN

### ISSUE THREAD

**Issue by HanbumKo** [ray-project/ray#15646]
### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*
`ray == 1.2.0`
`torch == 1.7.1`
`tensorflow == 2.4.1`
I have trained for 22hours but can't see any improvement of episode reward APEX-DQN on pong environment with tuned example [here](https://github.com/ray-project/ray/blob/master/rllib/tuned_examples/dqn/pong-apex.yaml)
Is there anything that I missed on code below?

### Reproduction (REQUIRED)

```
import ray
import ray.rllib.agents.dqn.apex as apex

ray.init()
config = apex.APEX_DEFAULT_CONFIG.copy()
config["framework"] = "torch"
config["target_network_update_freq"] = 20000
config["num_workers"] = 4
config["num_envs_per_worker"] = 8
config["lr"] = .00005
config["train_batch_size"] = 64
config["gamma"] = 0.99
trainer = apex.ApexTrainer(config=config, env="PongNoFrameskip-v4")

for i in range(1000000):
    result = trainer.train()
    print(result['episode_reward_mean'])

    if i % 100 == 0:

        checkpoint = trainer.save()
        print("checkpoint saved at", checkpoint)

```

<img width="1046" alt="Screen Shot 2021-05-05 at 5 39 19 PM" src="https://user-images.githubusercontent.com/37624747/117116239-1490d580-adc9-11eb-95f6-24292202d359.png">

**Comment by sven1977** [comment]
Hey @HanbumKo , could you try it with the above patch? I noticed that in torch, batch-indices for the prio replay buffer are converted by the SampleBatch to torch tensors and then sent via ray to the remote replay shards. On the way there, things don't get transferred correctly and the indices arrive all different, leading to wrong prioritizations in the buffer(s).
I'll try to confirm that this fixes the issue.

Also, for APEX, keep in mind that the epsilon exploration values are fixed(!) per worker and never change, such that it's probably always better to use evaluation workers to get an (explore=False) better evaluation (besides the data from the train batch, which has maybe too much exploration going on over the entire time). APEX does not reduce the epsilons over time, like vanilla DQN would do.

**Comment by HanbumKo** [comment]
Thanks, I've applied your commit and saw that worked well for the cartpole example. But, due to computation resource limitation, haven't tested Pong yet.
